server/controllers/cohort.js

```diff
@@ -126,13 +126,18 @@ const getCohortDetails = (req, resp) => {
         return resp.status(404).send({ message: 'Cohort data not found.' });
       }
 
-      const { _id, isArchived, name } = doc;
+      const { adminIds, _id, isArchived, description, name } = doc;
 
       if (isArchived) {
         return resp.status(200).json({ _id, isArchived, name });
       }
 
-      resp.status(200).json(doc);
+      const isAdmin = checkRole(adminIds, req.user._id);
+      const data = isAdmin
+        ? { _id, isAdmin, isArchived, description, name }
+        : { _id, isArchived, description, name };
```

Comment:
this seems bit confusing, you can always return the same object of data.
Once with `isAdmin` set as `true`, once as `false` and there will be no need of this ternary operator here. 

You need to propably change `checkRole` utlility method